daphne/src/messages/taskprov.rs

```diff
@@ -0,0 +1,293 @@
+// Copyright (c) 2022 Cloudflare, Inc. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+
+//! Messages in the taskprov extension to the DAP protocol, as
+//! defined in draft-wang-ppm-dap-taskprov-00.
+
+use crate::messages::{
+    decode_u16_bytes, encode_u16_bytes, Duration, Id, Time, QUERY_TYPE_FIXED_SIZE,
+    QUERY_TYPE_TIME_INTERVAL,
+};
+use prio::codec::{
+    decode_u16_items, decode_u24_items, decode_u8_items, encode_u16_items, encode_u24_items,
+    encode_u8_items, CodecError, Decode, Encode,
+};
+use ring::{
+    hmac::HMAC_SHA256,
+    hmac::{sign, Key},
+};
+use serde::{Deserialize, Serialize};
+use std::io::Cursor;
+
+// VDAF types defined in draft-wang-ppm-dap-taskprov-00
+const VDAF_TYPE_PRIO3_AES128_COUNT: u32 = 0x00000000;
+const VDAF_TYPE_PRIO3_AES128_SUM: u32 = 0x00000001;
+const VDAF_TYPE_PRIO3_AES128_HISTOGRAM: u32 = 0x00000002;
+const VDAF_TYPE_POPLAR1_AES128: u32 = 0x00001000; // The gap from the previous constant is intentional
+
+// Differential Privacy mechanism types defined in draft-wang-ppm-dap-taskprov-00
```

Comment:
(here and below) Now that taskprov things are in their own module, and the module refers to the draft, there's no reason to mention the draft in a comment.